api/v1alpha1/clustertemplateinstance_utils.go

```diff
@@ -292,3 +293,100 @@ func (i *ClusterTemplateInstance) GetHelmParameters(
 
 	return params, nil
 }
+
+func (i *ClusterTemplateInstance) GetSubjectsWithClusterTemplateUserRole(ctx context.Context, k8sClient client.Client) (subjects []rbacv1.Subject, err error) {
```

Comment:
Im not using named returns anywhere, so It would be good to align with the rest of the code.
Also the function does not actually take advantage of named returns
```suggestion
func (i *ClusterTemplateInstance) GetSubjectsWithClusterTemplateUserRole(ctx context.Context, k8sClient client.Client) ([]rbacv1.Subject,  error) {
```